### DISCUSSION TITLE
Email confirmation loop - simple design

### DISCUSSION THREAD

**Discussion by thomasiles** [govuk-forms/forms#46] [General]
# Submission email confirmation

## Introduction

[Trello card for this feature](https://trello.com/c/d4bc6tiC/344-front-end-build-the-confirmation-code-loop-for-setting-an-email-address-for-completed-forms-to-go-to-should-have)

As a form creator I want to be confident that the email address I set for email submissions is correct.

This feature adds a confirmation step to setting submission eamil when creating
forms. This should help ensure that the email is working and accessible for the
form creator.

Instead of directly editing the submission email address, a user must request
that a six digit code is sent to the email address. The user can then re-enter
this code to confirm that they have access to the email address.

Only one code per email is valid at any time per form.

Ability to clear the email address without setting a new one - maybe only in draft for the moment?

The admin will store the code, date sent, user who requested the code.

## Workflow

The roles of Admin and API.

```mermaid
sequenceDiagram
    Actor User
    participant Admin
    participant Notify
    participant API
    User->>Admin: Request code to email address
    Admin->>Notify: Send confirmation code
    Notify->>User: Email code to address
    User->>Admin: Enter confirmation code
    Admin->>API: Set Email Address
```
## Flow
A flow diagram showing the flow.

```mermaid
flowchart TD
    Start[Email blank] --> |User enters email address|Sent
    Sent[Email with code sent] --> |User enters code|Code
    Code --> |Code valid| Confirmed[Email set]
    Confirmed[Email set] --> |change email|Sent
    Confirmed[Email set] --> |clear email|Start
    Code{Code checked} --> |Code invalid| Start
```

**Comment by aliuk2012** [comment]
Does a code really need to expire after 7 days if it hasn't been used?
